synapse/http/matrixfederationclient.py

```diff
@@ -188,6 +188,60 @@ def schedule(x):
 
         self._cooperator = Cooperator(scheduler=schedule)
 
+    @defer.inlineCallbacks
+    def _send_request_with_optional_trailing_slash(
+        self,
+        request,
+        try_trailing_slash_on_400=False,
+        backoff_on_404=False,
```

Comment:
I don't think there's any need for this now. You can elide it and just let `send_request_args` pick it up.

Reply:
(if there are things that rely on passing it as a positional, rather than keyword, arg, then add a `*args` or, better, fix the callers to use the arg name)